cmd/neofs-adm/internal/modules/morph/initialize_register.go

```diff
@@ -39,18 +41,48 @@ func (c *initializeContext) registerCandidates() error {
 		return fmt.Errorf("can't fetch registration price: %w", err)
 	}
 
-	sysGas := regPrice + native.GASFactor // + 1 GAS
+	w := io.NewBufBinWriter()
+	emit.AppCall(w.BinWriter, neoHash, "setRegisterPrice", callflag.States, 1)
 	for _, acc := range c.Accounts {
-		w := io.NewBufBinWriter()
 		emit.AppCall(w.BinWriter, neoHash, "registerCandidate", callflag.States, acc.PrivateKey().PublicKey().Bytes())
 		emit.Opcodes(w.BinWriter, opcode.ASSERT)
+	}
+	emit.AppCall(w.BinWriter, neoHash, "setRegisterPrice", callflag.States, regPrice)
+	if w.Err != nil {
+		panic(fmt.Sprintf("BUG: %v", w.Err))
```

Comment:
is it gonna be kept like that?

Reply:
Added a little more context. We already have similar statements in `neofs-adm`.